network-store-client/src/main/java/com/powsybl/network/store/client/PreloadingNetworkStoreClient.java

```diff
@@ -124,6 +138,33 @@ private void ensureCached(ResourceType resourceType, UUID networkUuid, int varia
         }
     }
 
+    private void ensureExtensionAttributesCached(ResourceType resourceType, UUID networkUuid, int variantNum, String extensionName) {
+        Objects.requireNonNull(resourceType);
+        Objects.requireNonNull(networkUuid);
+        Objects.requireNonNull(extensionName);
+        Map<ResourceType, Set<String>> extensionAttributesByTypeAndName = cachedExtensionAttributesByTypeAndName.getCollection(networkUuid, variantNum);
+        Set<ResourceType> extensionAttributesByType = cachedExtensionAttributesByType.getCollection(networkUuid, variantNum);
+        boolean isFullyLoadedType = extensionAttributesByType.contains(resourceType);
+        boolean isFullyLoadedExtension = extensionAttributesByTypeAndName.containsKey(resourceType) && extensionAttributesByTypeAndName.get(resourceType).contains(extensionName);
+        if (!isFullyLoadedType || !isFullyLoadedExtension) {
```

Comment:
should be && ??

Reply:
missing a test ?

Reply:
You're correct. I inverted the condition but forgot half of it...
This bug is not present at runtime as there is a check later in the cache https://github.com/powsybl/powsybl-network-store/blob/591b34f742e03e6166706439abf734ed3cd20ecf/network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/CollectionCache.java#L429 which is correct this time. This prevents unnecessary fetches. I will fix this condition.

Reply:
Simplified this in https://github.com/powsybl/powsybl-network-store/pull/411/commits/1d2c57e75f0193c6d358372294460f2872ac1d4d 
It was already handled in CollectionCache.